jscomp/ml/typedecl.ml

```diff
@@ -479,6 +478,15 @@ let transl_declaration ~typeRecordAsObject env sdecl id =
           (match lbls_opt with
           | Some (lbls, lbls') ->
             check_duplicates lbls StringSet.empty;
+            let optionalLabels =
```

Comment:
Not sure any changes above this line are required?
Also the `let rep` above can probably be removed.

Reply:
From the description, the only logic requiring change is when splicing in new props.

Reply:
I reverted a few things - better?